google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -243,6 +245,25 @@ public void emptyQuery() throws Exception {
     assertEquals("bar", documents.next().get("foo"));
   }
 
+  @Test
+  public void queryForServerTimestamps() throws Exception {
+    DocumentReference documentReference = addDocument("serverTime", FieldValue.serverTimestamp());
+    DocumentSnapshot documentSnapshot = documentReference.get().get();
+    Timestamp serverTime = documentSnapshot.getTimestamp("serverTime");
+
+    if (TimeUnit.NANOSECONDS.toMicros(serverTime.getNanos()) > 0) {
+      // If serverTime has a microsecond component, this query produces no results as `getDate()`
```

Comment:
Hmm, if `serverTime.getNanos() == 1 * 1000 * 1000` (i.e., it's one millisecond), wouldn't the check above succeed? OTOH, it seems like the query should still succeed despite the truncation. I might misunderstand the code here, but it seems like the check above should be along the lines of `if (toMicros() % 1000 > 0`.

Reply:
You are correct that it was broken. With your fix, it becomes apparent that it is really broken - server timestamps only have millisecond precision.

I have removed the test and replaced it with "updateMicrosecondTimestamp", which verifies that microseconds don't get truncated.